language_server/src/server.h

```diff
@@ -64,11 +67,25 @@ class server : public response_provider
     bool exit_notification_received_ = false;
 
     parser_library::workspace_manager& ws_mngr_;
+    json_sink* telemetry_provider_;
 
     void register_feature_methods();
 
     // Calls a method that is registered in methods_ with the specified name with arguments and id of request.
     void call_method(const std::string& method, const json& id, const json& args);
+
+    struct telemetry_metrics_info
+    {
+        json properties;
+        json metrics;
+    };
+
+    telemetry_metrics_info virtual get_telemetry_details();
```

Comment:
This seems a bit strange... how is the method supposed to know to which request the details are being provided?

Reply:
I agree it is a bit strange... the details are provided for the last request.

It fetches the details for requests with `telemetry_log_level::LOG_WITH_METRICS`, which is actually only one request (didOpen).

I only wanted the `lsp_server` to have the metadata consumer, but at the same time I wanted to log the requests in the `server::call_method`, method, which is common for lsp and dap server. Thats why I ended up with this virtual method.